vg/recorder/recorder.go

```diff
@@ -0,0 +1,483 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package recorder provides support for vector graphics serialization
+// it is intended to be used as a testing helper.
+package recorder
+
+import (
+	"fmt"
+	"image/color"
+	"runtime"
+
+	"github.com/gonum/plot/vg"
+)
+
+var _ vg.Canvas = (*Canvas)(nil)
+
+// Canvas implements vg.Canvas operation serialization.
+type Canvas struct {
+	// Resolution holds the canvas resolution in DPI.
+	Resolution float64
+
+	// Actions holds a log of all methods called on
+	// the canvas.
+	Actions []Action
+
+	// KeepCaller indicates whether the Canvas will
+	// retain runtime caller location for the actions.
+	// This includes source filename and line number.
+	KeepCaller bool
+
+	// c holds a backing vg.Canvas. Method calls to
+	// the Canvas will be reflected to c.
+	c vg.Canvas
+
+	// fonts holds a collection of font/size descriptions.
+	fonts map[fontID]vg.Font
+}
+
+type fontID struct {
+	name string
+	size vg.Length
+}
+
+// Action is a vector graphics action as defined by the
+// vg.Canvas interface. Each method of vg.Canvas must have
```

Comment:
The second sentence sounds like an implementation detail. How about, "Each method of Canvas has a corresponding Action type."?